docs/contributing.rst

```diff
@@ -0,0 +1,121 @@
+Contributing
+============
+
+We are happy you have decided to contribute to ``twine``.
+
+Please see `the GitHub repository`_ for code and more documentation,
+and the `official Python Packaging User Guide`_ for user documentation. You can
+also join ``#pypa`` or ``#pypa-dev`` `on Freenode`_, or the `pypa-dev
+mailing list`_, to ask questions or get involved.
+
+Getting started
+---------------
+
+We recommend you use a development environment. Using a ``virtualenv``
+keeps your development environment isolated, so that ``twine`` and its
+dependencies do not interfere with packages already installed on your
+machine.  You can use `virtualenv`_ or `pipenv`_ to isolate your
+development environment.
+
+Clone the twine repository from GitHub, and then activate your virtual
+environment. Then, run the following command:
+
+.. code-block:: console
+
+  pip install -e /path/to/your/local/twine
+
+Now, in your virtual environment, ``twine`` is pointing at your local copy, so
+when you have made changes, you can easily see their effect.
+
+Testing
+^^^^^^^
+
+Tests with twine are run using `tox`_, and tested against the following Python
+versions: 2.7, 3.4, 3,5, and 3.6. To run these tests locally, you will need to
+have these versions of Python installed on your machine.
+
+If you are using ``pipenv`` to manage your virtual environment, you
+may need the `tox-pipenv`_ plugin so that tox can use pipenv
+environments instead of virtualenvs.
+
+Building the documentation
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Additions and edits to twine's documentation are welcome and appreciated. To
+build the docs locally, first set up a virtual environment and activate it,
+using Python 3.6 as the Python version in the virtual environment. Example:
+
+.. code-block:: console
+
+  mkvirtualenv -p /usr/bin/python3.6 twine
```

Comment:
It seems that users already have tox installed to run the tests. If that's the case, do they need a virtualenv to install it again?

Reply:
Good point. The guide here needed reorganization to get rid of this kind of duplication so I've now reordered and streamlined it.